File: Problems/No.2240_Number_of_Ways_to_Buy_Pens_and_Pencils/number_of_ways_to_buy_pens_and_pencils.cpp
class Solution {
public:
    long long waysToBuyPensPencils(int total, int cost1, int cost2) {
        
        // parameter
        long long cnt = 0;
        int higher_cost = std::max(cost1, cost2);
        int lower_cost = std::min(cost1, cost2);

        // start computing
        int remains = total;
        for (int i=0; i<=total/higher_cost; i++) {

            cnt += (remains/lower_cost + 1);
            remains -= higher_cost;

        }

        return cnt;

    }
};

File: Problems/No.2240_Number_of_Ways_to_Buy_Pens_and_Pencils/number_of_ways_to_buy_pens_and_pencils.py
class Solution(object):
    def waysToBuyPensPencils(self, total, cost1, cost2):
        """
        :type total: int
        :type cost1: int
        :type cost2: int
        :rtype: int
        """

        # parameters
        cnt = 0
        higher_cost = max(cost1, cost2)
        lower_cost = min(cost1, cost2)

        # start computing
        remains = total
        for i in range(total//higher_cost + 1):
            cnt += (remains//lower_cost + 1)
            remains -= higher_cost
        
        return cnt
